canutils/libdronecan/Makefile

```diff
@@ -0,0 +1,89 @@
+############################################################################
+# apps/canutils/libdronecan/Makefile
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+include $(APPDIR)/Make.defs
+
+WD := ${shell echo $(CURDIR) | sed -e 's/ /\\ /g'}
+
+UNPACK = unzip
+PACKEXT = .zip
+
+LIBDRONECAN_URL = $(patsubst "%",%,$(strip $(CONFIG_LIBDRONECAN_URL)))
+LIBDRONECAN_VERSION = $(patsubst "%",%,$(strip $(CONFIG_LIBDRONECAN_VERSION)))
+
+LIBDRONECAN_PACKNAME = $(LIBDRONECAN_UNPACKNAME)$(PACKEXT)
+LIBDRONECAN_UNPACKNAME = libcanard-$(LIBDRONECAN_VERSION)
+LIBDRONECAN_SRCNAME = libcanard
+
+LIBDRONECAN_SRCDIR = $(WD)/$(LIBDRONECAN_SRCNAME)
+LIBDRONECAN_DRVDIR = $(LIBDRONECAN_SRCDIR)$(DELIM)drivers$(DELIM)nuttx
+LIBDRONECAN_SOCKETCANDIR = $(LIBDRONECAN_SRCDIR)$(DELIM)drivers$(DELIM)socketcan
+
+APPS_INCDIR = $(APPDIR)$(DELIM)include$(DELIM)canutils
+
+# Conflict with Cyphal's libcanard
+ifeq ($(CONFIG_CANUTILS_LIBCANARDV1),y)
+APPS_INCDIR = $(APPDIR)$(DELIM)include$(DELIM)canutils$(DELIM)dronecan
+CFLAGS += -DcanardInit=dronecanardInit -I$(APPS_INCDIR)
+endif
+
+CFLAGS += -std=c99 -DCANARD_ASSERT=DEBUGASSERT
+CFLAGS += ${shell $(INCDIR) "$(CC)" $(APPS_INCDIR)}
+
+ifeq ($(CONFIG_LIBDRONECAN_CANFD),y)
+CFLAGS += -DCANARD_ENABLE_CANFD=1
+endif
+
+CSRCS = $(LIBDRONECAN_SRCDIR)$(DELIM)canard.c
+
+ifeq ($(CONFIG_NET_CAN),y)
+CSRCS += $(LIBDRONECAN_SOCKETCANDIR)$(DELIM)socketcan.c
+else
+CSRCS += $(LIBDRONECAN_DRVDIR)$(DELIM)canard_nuttx.c
+endif
+
+# Download and unpack tarball if no git repo found
+ifeq ($(wildcard $(LIBDRONECAN_SRCNAME)/.git),)
+$(LIBDRONECAN_PACKNAME):
+	@echo "Downloading: $@"
+	$(Q) curl -o $@ -L $(LIBDRONECAN_URL)$(DELIM)$(LIBDRONECAN_VERSION)$(PACKEXT)
```

Comment:
use DOWNLOAD macro

Reply:
I may have missed something, but where's this new DOWNLOAD macro documented?

Reply:
Please reference: https://github.com/apache/nuttx/pull/8759